File: Solution/WebSolution/AssessmentPortal/Monolithic/AssessmentServices/Implementations/EvaluationCriteriaService.cs
using MySql.Data.MySqlClient;
using System.Data;
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Services.Interfaces;

namespace Transflower.TFLAssessment.Services.Implementations;

//Providers
public class EvaluationCriteriaService : IEvaluationCriteriaService
{
    private readonly IEvaluationCriteriaRepository _repository;

    public EvaluationCriteriaService(IEvaluationCriteriaRepository repository)
    {
        _repository = repository;

    }
    public List<EvaluationCriteria> GetEvalutionCriterias()
    {
        return _repository.GetEvalutionCriterias;

    }

    public List<EvaluationCriteria> GetEvalutionCriteriasBySubject(int subjectId)
    {
        return _repository.GetEvalutionCriteriasBySubject(subjectId);
    }

    public string GetCriteria(string subject, int questionId)
    {
        return _repository.GetCriteria(subject, questionId);
    }

    public bool InsertCriteria(NewCriteria criteria)
    {
        return _repository.InsertCriteria(criteria);
    }

    public bool UpdateCriteria(int evaluationCriteriaId, int questionId)
    {
        return _repository.UpdateCriteria(evaluationCriteriaId, questionId);
    }
}

File: Solution/WebSolution/Java/AssessmentAPI/src/main/java/com/transflower/tflassessment/services/EvaluationCriteriaService.java
package com.transflower.tflassessment.services;

import java.util.concurrent.CompletableFuture;

import org.springframework.stereotype.Service;

import com.transflower.tflassessment.entities.EvaluationCriteria;

@Service
public interface  EvaluationCriteriaService {
     
    public CompletableFuture<Boolean> updateSubject(int id, int subjectId);
 
    public CompletableFuture<Boolean> insertCriteria(EvaluationCriteria ec);

    public CompletableFuture<Boolean> updateCriteria(int EvaluationCriteriaId, int subjectId);

}
